collectors/ebpf.plugin/ebpf.c

```diff
@@ -513,44 +516,68 @@ int ebpf_exit_plugin = 0;
  */
 static void ebpf_stop_threads(int sig)
 {
-    ebpf_exit_plugin = 1;
+    UNUSED(sig);
+    static int only_one = 0;
+    // Child thread should be closed by itself.
+    pthread_mutex_lock(&ebpf_exit_cleanup);
+    if (main_thread_id != gettid() || only_one) {
+        pthread_mutex_unlock(&ebpf_exit_cleanup);
+        return;
+    }
+    only_one = 1;
+    pthread_mutex_unlock(&ebpf_exit_cleanup);
+
     int i;
-    for (i = 0; ebpf_threads[i].name != NULL; i++);
+    pthread_mutex_lock(&ebpf_exit_cleanup);
```

Comment:
Any particular reason why it unlocks, then locks again after? Could it keep the lock up until 536?

Reply:
My bad, probably it was a commit during the dawn time. I moved the code, thanks!